### ISSUE TITLE
Cutwork Work flow

### ISSUE THREAD

**Issue by owendelong** [inkstitch/inkstitch#1748]
This is mostly a feature request.

I use a Barudan 15 needle commercial machine with Inkstitch (THANK YOU!! for your FDR/.U01 file support).

For cutwork, it has an omnidirectional "boring" needle, so I don't need to segment by angle. However, what I would like to see is an easier work flow for setting up the actual cut+stitching combinations that are needed to make useful cutwork.

Ideally, I'd like to be able to define a closed-shape outline path that defines the outer edge of the cut area and then "convert path to cutwork". This should result in an appropriate set of interior cut paths (line, X, or line + cross-lines) that will free the fabric for stitching, then a set of wide-spread satin stitches that go from the proximate cut line to a point 2-3mm beyond the cut edge. (These will pull the fabric to the edge of the cut area and wrap it in thread to retain it near the edge). Then finally, a satin column of appropriate width should be created with the inner rail just inside the cut area and the outer rail a distance from the edge specified by the width of the path (similar to convert line to satin).

Admittedly, this is based on some web research and I have not yet successfully digitized good cutwork (since I'm using InkStitch to do my digitizing it is proving difficult to get good results in the current program).

I am not new to embroidery, but I am relatively new to InkScape, InkStitch, and very new to cutwork embroidery.

**Comment by lexelby** [comment]
This sounds really great.  I haven't had a chance to play with one of these commercial boring needles.  The one for my old machine would have cost $1500+.

I can't really follow your description.  It would really help to have a diagram or perhaps a video example of a stitch-out.  There are definitely things I don't know about the principles involved (how to avoid shreds of fabric gunking up your rotary hook or under the needle plate, etc).

Actually the thing that would greatly help would be if you did the experimentation necessary to figure out a good process, and then posted an example SVG.  So that would mean an example design where you do the things you describe above "by hand" with running stitch and satin objects.  Then we could turn that into code to emulate your process automatically for an arbitrary shape.

I know I'm asking a lot of work from you, and it's fine if you can't do it. :)  It would really help the project if you can though, because none of us has an omnidirectional boring needles to test with.  

@wwderw might be interested in this as he has a similar Barudan.  He's the one that helped us develop the u01 support.  I think his machine has a sequin attachment where yours has the boring needle though.

**Comment by wwderw** [comment]
Yes, my Barudan is with the sequin attachment.  Which was able to get that to work via export to CSV and manually put in the sequin drop functions on the stitches where applicable and take out the stitch function.  There may be something like that, that could be done for the boring functionality.

The downside to this approach is that if you make an edit, you would have to re-export to the CSV and than convert back to a U?? format for embroidery.  But it would (if my assumption is correct in how this can be achieved) give you this functionality, if those functions discovered.  STITCH for stitching etc.  That would need to be determined in order to do this.

While that may seem tedious, I have found that using the stitch counter in the simulator to give me a good start part and end point for where I need to do my function changes manually and I have them in a todo layer in Inkscape, so that way it's always there with the master file in case I go back to the project months, years down the line and I have forgotten what the file has incorporated in it. 

I do have somewhat of a boring functionality on my Brother machine, but it is angle based, in other words, I have multiple needles with different angle teeth on them and depending on what part of the cut path it's doing, determines what needle is used.  Not something that would help with testing the above boring functionality as on full blown commercial machines it is one needle.  It's been awhile since I have done cutwork with the embroidery machine, that was always done last if I recall, that may or may not be the same as with the Barudan, but if it is, that would cut down on the how tedious it is to manually insert those commands.

**Comment by lexelby** [comment]
@wwderw you know the latest version has a new extension to work with those brother cutwork needle sets, right?

The boring attachment that the OP has won't need csv editing. I believe those use normal STITCH commands, they just have to be offset a bit.

**Comment by kaalleen** [comment]
I think this feature request is very similar to this (old) one for applique functionality (just this time for it is for cutwork): #112

Appliqué:

> Placement stitch, tack down stitch (x2) and a cover stitch.

Cutwork (e.g.):

* Placement stitch
* Cut stitch
* Cover stitch (zig-zag tack down (underlay), satin finish)

**Comment by owendelong** [comment]
I haven't been able to fully duplicate this particularly well yet, but here's a video showing what I'm getting at:
https://www.youtube.com/watch?v=LBecb3hqB9M

**Comment by owendelong** [comment]
I agree with Kaalleen that this is likely similar to the process for appliqué. FWIW, my Brother luminaire integrated with the Brother Scan'n'Cut actually has a workflow worth imitating for the appliqué. It can produce a slightly reduced-scale outline stitch on the hooped fabric for appliqué placement. It will also produce a cut file for the scan'n'cut at the right scale and an appropriate satin column for stitching down the appliqué (which extends from the placement line stitch to just outside of the appliqué. (Yes, the luminaire is a different machine from the Barudan that I created this issue for).

The technique that comes closest to ideal results so far for me on the cutwork is as follows:

1. outline the area to be cut with a double perimeter of straight stitches offset approximately 0.5mm from the cut area.
2. raster cut out the cut area. (perimeter cutting results in a chunk of fabric that launches into a random location)
3. run a satin column from approximately 0.25mm inside of the cut perimeter to just outside of the straight stitch perimeter (or any later desired width) around the perimeter of the cutwork.

I haven't managed to get sufficient control of the alignment to result in particularly even placement (the cut fabric shifts in unreliable directions or folds over in unexpected and uneven ways), but I'm still trying experiments to improve the tuning of this.

**Comment by tyrosinase** [comment]
I'm going to drop this here and run because I administer a Mastodon instance and things are getting SLAMMED right now, but here's a wholly-manual workflow as a FWIW: https://silverseams.com/tutorials/making-appliques-with-ink-stitch/

**Comment by owendelong** [comment]
> On Jul 23, 2022, at 07:04, wwderw ***@***.***> wrote:
> 
> 
> Yes, my Barudan is with the sequin attachment. Which was able to get that to work via export to CSV and manually put in the sequin drop functions on the stitches where applicable and take out the stitch function. There may be something like that, that could be done for the boring functionality.
> 
I don’t think that’s necessary for boring functionality. An ordinary raster fill (fill stitch with 0.5mm stitch length) works just fine for the actual cutting (turns the area to be cut into confetti, requires more frequent cleaning of the bobbin area).

I’m pretty sure that the straight-stitch circumference followed by boring (raster) followed by satin column process I outlined is the right one. I just need to figure out the limits of the machines consistency/reproducability and then come up with spacing on the stitches that is best adapted to those parameters.
> The downside to this approach is that if you make an edit, you would have to re-export to the CSV and than convert back to a U?? format for embroidery. But it would (if my assumption is correct in how this can be achieved) give you this functionality, if those functions discovered. STITCH for stitching etc. That would need to be determined in order to do this.
> 
Which is why I think that is best avoided.

The boring needle is a change to the MC parameters to tell the machine that needle X is a boring needle. No change is needed in the stitching file, you just need to map the cuts to the needle number where the boring needle is installed. It doesn’t require any special functionality from the Barudan the way that the sequins installer does. It doesn’t even disable the “stitching”, just the thread break detection since no thread will be moving.
> While that may seem tedious, I have found that using the stitch counter in the simulator to give me a good start part and end point for where I need to do my function changes manually and I have them in a todo layer in Inkscape, so that way it's always there with the master file in case I go back to the project months, years down the line and I have forgotten what the file has incorporated in it.
> 
> I do have somewhat of a boring functionality on my Brother machine, but it is angle based, in other words, I have multiple needles with different angle teeth on them and depending on what part of the cut path it's doing, determines what needle is used. Not something that would help with testing the above boring functionality as on full blown commercial machines it is one needle. It's been awhile since I have done cutwork with the embroidery machine, that was always done last if I recall, that may or may not be the same as with the Barudan, but if it is, that would cut down on the how tedious it is to manually insert those commands.
> 
FWIW, a set of 10 boring needles for the Barudan is relatively cheap. Heck, if you want, I can probably send you a couple of mine which are still very lightly used and I can’t imagine I’ll ever go through more than 8 of them in the foreseeable future.

Owen
